### ISSUE TITLE
pandas Series.reset_index() returning wrong type when index of Series is a MultiIndex

### ISSUE THREAD

**Issue by Dr-Irv** [microsoft/pylance-release#393]
-   Language Server version: 2020.9.5
-   OS and version: Windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): Anaconda Python 3.7.5

## Expected behaviour

No error

## Actual behaviour

```python
mi3: pd.Index = pd.MultiIndex.from_product(
    [[1, 2], [3, 4], [5, 6]], names=["a", "b", "c"]
)
s = pd.Series([i for i in range(len(mi3))], index=mi3)
dr: pd.DataFrame = s.reset_index(-1)
print("dr")
print(dr)
print(type(dr))
```

reports for line 5
```
Expression of type "Series[Unknown]" cannot be assigned to declared type "DataFrame"
  "Series[Unknown]" is incompatible with "DataFrame"
```

The issue here is that `reset_index()` can return a `Series` or `DataFrame` dependent on whether the parameter `drop` is `True` or `False`. (It is `False` by default).  In the example above, it is returning a `DataFrame`

**Comment by gramster** [comment]
Thanks for all these reports and suggested fixes; that's awesome!

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.9.6, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202096-23-september-2020
